components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/context/OptimizedSessionContext.java

```diff
@@ -88,11 +88,14 @@ public OptimizedSessionContext(SessionContext sessionContext) throws SessionData
                     for (Map.Entry<String, IdentityProvider> idpEntry : authenticatorConfig.getIdps().entrySet()) {
                         IdentityProvider idp = idpEntry.getValue();
                         String idpName = idpEntry.getKey();
-                        if (idp.getResourceId() == null) {
+                        if (StringUtils.isEmpty(idp.getResourceId())) {
                             idPResourceId.add(getIdPByIdPName(idpName, this.tenantDomain).getResourceId());
                         } else {
                             idPResourceId.add(idp.getResourceId());
                         }
+                        if (StringUtils.isEmpty(authenticatorConfig.getTenantDomain())) {
+                            authenticatorConfig.setTenantDomain(this.tenantDomain);
```

Comment:
here the `tenantDomain` should be taken from `idp`

Reply:
Resolved with [`8ae0557`](https://github.com/wso2/carbon-identity-framework/pull/4446/commits/8ae05575266f68d589e548a7dc82bf625e079337)